DATACOUCH-91 - Support auditing (via java config)

simonbasle opened the pull request:
This commit adds support for auditing via the use of the four annotations (CreatedBy, LastModifiedBy, CreationDate, LastModificationDate), enabled via the EnableCouchbaseAuditing annotation.

As couchbase entities MUST have an id, the only case where the creation event can be detected is when the entity has a Version annotated field.

Pinging @olivergierke @daschl for review :smile:

daschl commented:
I don't think I'll be a good reviewer here since I don't know much about the auditing stuff. Deferring to @olivergierke if he has a minute or two

simonbasle commented:
added documentation and rebased on latest master. @olivergierke if you have time to review this today it'll be great, otherwise I'll go ahead and +1 myself for now.

simonbasle commented:
:+1: let's get back to it in a separate change if anything comes up
